.github/workflows/maven-publish.yml

```diff
@@ -27,4 +28,19 @@ jobs:
           MAVEN_USERNAME: ${{ secrets.OSSRH_USERNAME }}
           MAVEN_PASSWORD: ${{ secrets.OSSRH_TOKEN }}
         run: mvn --batch-mode -Dgpg.passphrase=${{ secrets.OSSRH_GPG_SECRET_KEY_PASSWORD }} clean deploy
-    
+
+      - name: Guess Dependency Declaration Version
+        run: |
+          cd .github
+          javac ReadmeUpdater.java
+          java ReadmeUpdater ${{github.ref_name}}
+          rm -f ReadmeUpdater.class
+      - name: Propose a PR to Update Readme
```

Comment:
It seems a PR will be created anyway, even if the readme in main branch is already updated manually?

Reply:
If the file is already satisfied, no changes will be made by the Java program, thus *git add .* actually track no change and it won't raise a PR, as tested.